src/libstore/build/derivation-goal.cc

```diff
@@ -292,22 +292,28 @@ void DerivationGoal::haveDerivation()
 
     parsedDrv = std::make_unique<ParsedDerivation>(drvPath, *drv);
 
-
     /* We are first going to try to create the invalid output paths
        through substitutes.  If that doesn't work, we'll build
        them. */
     if (settings.useSubstitutes && parsedDrv->substitutesAllowed())
-        for (auto & [_, status] : initialOutputs) {
-            if (!status.wanted) continue;
-            if (!status.known) {
-                warn("do not know how to query for unknown floating content-addressed derivation output yet");
-                /* Nothing to wait for; tail call */
-                return DerivationGoal::gaveUpOnSubstitution();
+        for (auto& [outputName, status] : initialOutputs) {
+            if (!status.wanted)
+                continue;
+            auto drvSubstitution = getEnv("USE_DRV_SUBSTITUTION");
```

Comment:
Is it possible to use a setting instead? The proliferation of environment variables that changes the behaviour of Nix can be problematic.